be/src/exec/hdfs_scanner.cpp

```diff
@@ -104,17 +104,31 @@ Status HdfsScanner::_build_scanner_context() {
         partition_values.emplace_back(std::move(partition_value_column));
     }
 
+    ctx.conjunct_ctxs_by_slot = _scanner_params.conjunct_ctxs_by_slot;
+
     // build columns of materialized and partition.
     for (size_t i = 0; i < _scanner_params.materialize_slots.size(); i++) {
         auto* slot = _scanner_params.materialize_slots[i];
 
-        HdfsScannerContext::ColumnInfo column;
-        column.slot_desc = slot;
-        column.idx_in_chunk = _scanner_params.materialize_index_in_chunk[i];
-        column.decode_needed =
-                slot->is_output_column() || _scanner_params.slots_of_mutli_slot_conjunct.find(slot->id()) !=
-                                                    _scanner_params.slots_of_mutli_slot_conjunct.end();
-        ctx.materialized_columns.emplace_back(std::move(column));
+        // if `can_use_any_column`, we can set this column to non-existed column without reading it.
+        if (_scanner_params.can_use_any_column && slot->col_name() != "___count___") {
+            ctx.not_existed_slots.push_back(slot);
+            SlotId slot_id = slot->id();
+            if (ctx.conjunct_ctxs_by_slot.find(slot_id) != ctx.conjunct_ctxs_by_slot.end()) {
```

Comment:
How about extracting a static util function to remove the none-existed slot_id in `conjunct_ctxs_by_slot`

this code is duplicated with `update_materialized_columns()`

Reply:
done